libfs/fs.go

```diff
@@ -96,15 +110,41 @@ func NewFS(ctx context.Context, config libkbfs.Config,
 	if len(subdir) > 0 {
 		parts = strings.Split(subdir, "/")
 	}
-	for i, p := range parts {
-		n, ei, err = config.KBFSOps().Lookup(ctx, n, p)
-		if err != nil {
-			return nil, err
-		}
-		if ei.Type != libkbfs.Dir {
-			return nil, errors.Errorf("%s is not a directory",
-				path.Join(parts[:i]...))
+	// Loop while we follow symlinks.
+outer:
+	for {
```

Comment:
I just didn't want to have to modify the `parts` slice and the iteration index, during iteration.  Seems a little harder to get right, and this should be fast enough since everything will be cached after the first iteration.  I could be convinced to try it the other way if you really prefer it, though.

Reply:
Got you. That's OK; I didn't think through the whole thing for the other approach, so if you think this way it's easier to get right then let's stick with this!